neo/Ledger/MemPool.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Neo.Network.P2P.Payloads;
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+
+namespace Neo.Ledger
+{
+    internal class MemPool : IReadOnlyCollection<Transaction>
+    {
+        private class PoolItem
+        {
+            public readonly Transaction Transaction;
+            public readonly DateTime Timestamp;
+
+            public PoolItem(Transaction tx)
+            {
+                Transaction = tx;
+                Timestamp = DateTime.UtcNow;
+            }
+        }
+
+        private const int MemoryPoolSize = 50000;
+        private readonly ConcurrentDictionary<UInt256, PoolItem> mem_pool = new ConcurrentDictionary<UInt256, PoolItem>();
+
+        public int Count => mem_pool.Count;
+
+        public void Clear()
+        {
+            mem_pool.Clear();
+        }
+
+        public bool ContainsKey(UInt256 hash)
+        {
+            return mem_pool.ContainsKey(hash);
+        }
+
+        public IEnumerator<Transaction> GetEnumerator()
+        {
+            return mem_pool.Select(p => p.Value.Transaction).GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();

```

Comment:
I will test that this pointed to https://github.com/neo-project/neo/pull/392/files#diff-692a215303b3871f385ba1044c5fe5feR40 ?

Reply:
PASS :)